src/bika/lims/browser/worksheettemplate.py

```
@@ -31,16 +29,20 @@ class AJAXGetWorksheetTemplateInstruments(BrowserView):
     Returns a vocabulary with the instruments available for the selected method
     """
     def __call__(self):
-        if 'method_uid' in self.request.keys():
-            method_uid = str(json.loads(self.request.get('method_uid', '')))
-            cfilter = {
-                'portal_type': 'Instrument',
-                'is_active': True,
-                'getMethodUIDs': {"query": method_uid,
-                                  "operator": "or"}}
-            bsc = getToolByName(self, 'senaite_catalog_setup')
-            items = [{'uid': '', 'm_title': 'No instrument'}] + [
-                {'uid': o.UID, 'm_title': o.Title} for o in
-                bsc(cfilter)]
-            items.sort(lambda x, y: cmp(x['m_title'], y['m_title']))
+        items = [{'uid': '', 'm_title': 'No instrument'}]
+        if 'method_uid' not in self.request.keys():
             return json.dumps(items)
+
+        method_uid = str(json.loads(self.request.get('method_uid', '')))
+        if not method_uid:
+            return json.dumps(items)
+
+        try:
+            instruments = api.get_object(method_uid).getInstruments()
+        except api.APIError:
```

Better remove this try/except. System might fail silently otherwise.